examples/developer-tools/src/blocks/inputs.ts

```diff
@@ -0,0 +1,149 @@
+import * as Blockly from 'blockly';
+
+const fieldMessage = 'fields %1 %2';
+const fieldArgs = [
+  {
+    'type': 'field_dropdown',
+    'name': 'ALIGN',
+    'options': [['left', 'LEFT'], ['right', 'RIGHT'], ['centre', 'CENTRE']],
+  },
+  {
+    'type': 'input_statement',
+    'name': 'FIELDS',
+    'check': 'Field',
+  },
+];
+
+const typeMessage = 'type %1';
+const typeArgs = [
+  {
+    'type': 'input_value',
+    'name': 'TYPE',
+    'check': 'Type',
+    'align': 'RIGHT',
+  },
+];
+
+/**
+ * Check to see if more than one input has this name.
+ * Highly inefficient (On^2), but n is small.
+ * @param referenceBlock Block to check.
+ */
+function inputNameCheck(referenceBlock: Blockly.Block) {
+  if (!referenceBlock.workspace) {
+    // Block has been deleted.
+    return;
+  }
+  const name = referenceBlock.getFieldValue('INPUTNAME').toLowerCase();
+  let count = 0;
+  const blocks = referenceBlock.workspace.getAllBlocks(false);
+  for (const block of blocks) {
+    const otherName = block.getFieldValue('INPUTNAME');
+    if (block.isEnabled() && !block.getInheritedDisabled() &&
+        otherName && otherName.toLowerCase() === name) {
+      count++;
+    }
+  }
+  const msg = (count > 1) ?
+      'There are ' + count + ' input blocks\n with this name.' : null;
+  referenceBlock.setWarningText(msg);
+}
+
+export const inputValue = {
+  // Value input.
+  init: function() {
+    this.jsonInit({
+      'message0': 'value input %1 %2',
+      'args0': [
+        {
+          'type': 'field_input',
+          'name': 'INPUTNAME',
+          'text': 'NAME',
+        },
+        {
+          'type': 'input_dummy',
+        },
```

Comment:
I don't think the trailing dummy input is necessary. It should get added automatically. Although maybe that doesn't work with multiple message strings?